src/cloudWatchLogs/registry/logStreamRegistry.ts

```diff
@@ -258,7 +277,7 @@ export type CloudWatchLogsAction = (
     logGroupInfo: CloudWatchLogsGroupInfo,
     apiParameters: CloudWatchLogsParameters,
     nextToken?: string
-) => Promise<CloudWatchLogsResponse>
+) => Promise<CloudWatchLogsResponse | void>
```

Comment:
This seems like a small change, but we basically just doubled the complexity by doing this. The lower level your logic is, the more costly this becomes.